app/apollo/resolvers/channel.js

```diff
@@ -208,7 +211,10 @@ const channelResolvers = {
         if(!channel){
           throw new NotFoundError(context.req.t('Could not find the configuration channel with uuid/name {{channelUuid}}/channelName.', {channelUuid}), context);
         }
+
+        logger.info({req_id, user, org_id, channelUuid, channelName}, `${queryName} validating`);
         await validAuth(me, org_id, ACTIONS.READ, TYPES.CHANNEL, queryName, context, [channel.uuid, channel.name]);
+        logger.info({req_id, user, org_id, channelUuid, channelName}, `${queryName} validating - authorized`);
```

Comment:
- The `validating` log message should be at the beginning of the validation section (i.e. right after the `try {` line).
- Simple checks for valid input should be first
- While the DB query can be before the `validAuth` call, the throwing of NotFoundError should be _after_, to avoid information leakage.
- Similarly it shouldn't return "more than one" error until after it has determined if the user is authorized or not -- if the user isn't authorized on what they are querying, we shouldn't tell them there exist multiple.
- The validAuth call will need to change it's `[ id, name ]` argument depending on whether the cluster was found in the DB or not.